thrift_rpc/recorder.go

```diff
@@ -385,6 +387,19 @@ func (r *Recorder) Flush() {
 	}
 }
 
+func (r *Recorder) Close() error {
+	r.lock.Lock()
+	closech := r.closech
+	r.closech = nil
+	r.lock.Unlock()
+
+	if closech != nil {
+		close(closech)
+	}
+
+	return nil
```

Comment:
I was looking at using this new API and I second @aybabtme's observation. I don't understand your reply as this isn't specific to thrift. The main usecases of a cleanup API is to make sure the last batch of traces make it before exiting (e.g. `defer CloseTracer(t)` in `main`. Without blocking until the final flush completes, this API is not useful for that.